### PR TITLE
Fix compilation due to powsybl-core 3.5.0 breaking changes

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
N/A

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Update powsybl-core version

**What is the current behavior?** *(You can also link to an open issue here)*
The project depends on powsybl-core v3.4.0

**What is the new behavior (if this is a feature change)?**
The project depends on powsybl-core v3.5.0-SNAPSHOT and the compilation issues are fixed to be ready for the next release.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by mathbagu** [comment]
**TODO**: update powsybl-single-line-diagram version (next release) and fix powsybl-core version to 3.5.0 when the release is available

@geofjamg: I put a fixme in the code about the management of non linear shunt. As the type of the feeder can change depending on the number of section enabled, I suppose there is something more clever to do to change the icon associated to the feeder, but I'm not sure the code is ready for that, or how to do that. Let me know if it's OK

I put this change to help Konan to test powsybl-core candidate release.

**Comment by geofjamg** [comment]
> 
> 
> **TODO**: update powsybl-single-line-diagram version (next release) and fix powsybl-core version to 3.5.0 when the release is available
> 
> @geofjamg: I put a fixme in the code about the management of non linear shunt. As the type of the feeder can change depending on the number of section enabled, I suppose there is something more clever to do to change the icon associated to the feeder, but I'm not sure the code is ready for that, or how to do that. Let me know if it's OK
> 
> I put this change to help Konan to test powsybl-core candidate release.

The number of activated sections cannot change the sign of the inductance?

**Comment by mathbagu** [comment]
> The number of activated sections cannot change the sign of the inductance?

For a non linear model, I think it's possible to have sections with a negative B and sections with a positive B. I'm not totally sure it's allowed, but if it is, the sign could change

File: single-line-diagram-core/src/main/java/com/powsybl/sld/NetworkGraphBuilder.java [hunk 1]
```diff
@@ -126,8 +126,9 @@ private FeederNode createFeederNode(Graph graph, Injection injection) {
                 case STATIC_VAR_COMPENSATOR:
                     return FeederInjectionNode.createStaticVarCompensator(graph, injection.getId(), injection.getName());
                 case SHUNT_COMPENSATOR:
-                    return ((ShuntCompensator) injection).getbPerSection() >= 0 ? FeederInjectionNode.createCapacitor(graph, injection.getId(), injection.getName())
-                                                                                : FeederInjectionNode.createInductor(graph, injection.getId(), injection.getName());
+                    // FIXME(mathbagu): Non linear shunt can be capacitor or inductor depending on the number of section enabled
```
Comment (jonenst) [inline single-line-diagram-core/src/main/java/com/powsybl/sld/NetworkGraphBuilder.java:129]: should we fix this before merging ?

Comment (geofjamg) [reply single-line-diagram-core/src/main/java/com/powsybl/sld/NetworkGraphBuilder.java:129]: No

**Comment by jonenst** [comment]
Sonar issues can be fixed later

**Pull request merged**